crates/bindings-csharp/Runtime/Internal/ITable.cs

```diff
@@ -7,12 +7,18 @@ public interface ITable<T> : IStructuralReadWrite
     where T : ITable<T>, new()
 {
     // These are the methods that codegen needs to implement.
-    void ReadGenFields(BinaryReader reader);
-    static abstract TableDesc MakeTableDesc(ITypeRegistrar registrar);
+    static abstract IEnumerable<TableDesc> MakeTableDesc(ITypeRegistrar registrar);
+
     static abstract Filter CreateFilter();
+}
 
-    // These are static helpers that codegen can use.
+public interface ITableView<View, T>
+    where View : ITableView<View, T>
+    where T : ITable<T>, new()
+{
+    static abstract T ReadGenFields(BinaryReader reader, T row);
```

Comment:
We can leave this as a future cleanup, but it seems like it would be easier to make this a regular method on `ITable<T>` instead of `static abstract` on `ITableView`? Since it operates on `row` instance anyway.

Reply:
That was my first approach, but with each table view able to have a different set of unique keys, they become per-view functions.